### PR TITLE
fix: Remove all 404s, make linkchecker soft-required

### PR DESCRIPTION
**Pull request by untitaker**
Anchors are still ignored.

A lot of the "fixes" I am doing here are already handled by redirects in vercel.json, but to get some safety guards in CI going I fixed those too.

If the linkchecker is not too red all the time let's make it required for merge at a later point (maybe)

### CODE REVIEW

File: src/platforms/common/enriching-events/attachments/index.mdx [hunk 1]
```diff
@@ -10,14 +10,13 @@ notSupported:
   - native.breakpad
   - native.crashpad
   - native.minidumps
-  - native.ue4
```
Comment (untitaker) [inline src/platforms/common/enriching-events/attachments/index.mdx:13]: cc @jan-auer, looking at comments this seems to have never been intended https://github.com/getsentry/sentry-docs/pull/2343

File: src/platforms/native/guides/minidumps/index.mdx [hunk 2]
```diff
@@ -116,7 +116,7 @@ documents.
 ## Event Attachments
 
 Besides the minidump file, Sentry can optionally store additional files uploaded
-in the same request, such as log files. Learn more about these in [Attachments](enriching-error-data/attachments/).
```
Comment (untitaker) [inline src/platforms/native/guides/minidumps/index.mdx:119]: cc @jan-auer, this page is disabled for minidumps, see https://github.com/getsentry/sentry-docs/pull/2343

**Review by dcramer — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -101,4 +101,8 @@ jobs:
         run: |-
           http-server public/ > /dev/null &
           printf "Waiting for server to be up"; timeout 60 bash -c 'until $(curl -Isf -o /dev/null "http://localhost:8080"); do printf '.'; sleep 0.5; done'
-          ./linkcheck http://localhost:8080 || true
+
+          # exitcode=1 -- only warnings (about broken anchors)
+          # exitcode=2 -- hard 404s
+          # We only fail the build hard 404s
```
Comment (untitaker) [inline .github/workflows/test.yml:107]:
```suggestion
          # We only fail the build if there are hard 404s
```

File: src/platforms/native/guides/minidumps/index.mdx (hunk 2 above)
Comment (jan-auer) [reply src/platforms/native/guides/minidumps/index.mdx:119]: I tried to re-enable it on Friday, but Gatsby wouldn't let me. Maybe you can help me with that? I'd like to keep the link in.

**Review by jan-auer — APPROVED** [review]

**Comment by untitaker** [comment]
Following up with @jan-auer on native page in private.

**Pull request merged**